src/modules/dataProviders/providers/squid/squid.helper.ts

```diff
@@ -0,0 +1,46 @@
+import { Inject, Injectable, OnApplicationBootstrap } from '@nestjs/common';
+import { Client as GraphqlWsClient } from 'graphql-ws';
+import { NotificationService } from '../../../notification/services/notification.service';
+import {
+  SquidContentExtensionSchemaId,
+  SquidActivitiesResponseDto,
+  SquidNotificationsResponseDto,
+  SquidSubscriptionResponseDto
+} from '../../dto/squid/squidResponse.dto';
+import { SquidApiQueryName } from '../../typeorm/squidDataSubscriptionStatus';
+import { DataProvidersService } from '../../services/dataProviders.service';
+import { EventName } from '../../dto/squid/squidEvents.dto';
+import axios from 'axios';
+import { xSocialConfig } from '../../../../config';
+
+type SquidApiResponse<Q, R> = Q extends SquidApiQueryName.notifications
+  ? {
+      notifications: Array<SquidNotificationsResponseDto>;
+    }
+  : Q extends SquidApiQueryName.activities
+  ? { activities: SquidActivitiesResponseDto }
+  : never;
+
+@Injectable()
+export class SquidHelper {
+  constructor(
+    @Inject('GraphqlWsClient') private graphqlWsClient: GraphqlWsClient,
```

Comment:
You call it as WsClient and use HTTP_URL with axios, so maybe Ws is useless here?

Reply:
It's redundant Injection. Removed.